src/main/java/seedu/address/ui/WidgetViewBox.java

```diff
@@ -1,108 +1,141 @@
-        divOne.setText(Locale.getDefault().getDisplayCountry());
-        textOne.setText(TimeZone.getDefault().getDisplayName());
-        divTwo.setText("Travelling BusinessMan");
-        textTwo.setText("");
-        textThree.setText("");
-        divThree.setText("");
-        textFour.setText("");
-        footer.setText("Made in NUS");
+        this.clientObservableList = clientObservableList;
+
+        clientObservableList.addListener((ListChangeListener<Client>) c -> {
+            if (c.next()) {
+                if (c.wasUpdated() || c.wasReplaced()) {
+                    updateClientDisplay(clientObservableList.get(displayedClientIndex));
+                } else {
+                    for (int i = 0; i < clientObservableList.size(); i++) {
+                        Client client = clientObservableList.get(i);
+                        if (client.isSameClient(displayedClient)) {
+                            displayedClient = client;
+                            displayedClientIndex = i;
+                            return;
+                        }
+                    }
+                }
+            }
+        });
+        initDefault();
     }
 
     /**
-     * Private constructor for testing purposes.
+     * Updates the current content of the widget view box to the given content.
      *
-     * @param object Widget Object to be used in testing.
+     * @param client The new content.
      */
-    private WidgetViewBox(WidgetObject object) {
-        super(FXML);
-        this.widgetObject = object;
-        textClock = new TextClock(header);
-        header.setText(object.getHeader());
-        divOne.setText(object.getDivOne());
-        textOne.setText(object.getTextOne());
-        divTwo.setText(object.getDivTwo());
-        textTwo.setText(object.getTextTwo());
-        textThree.setText(object.getTextThree());
-        divThree.setText(object.getDivThree());
-        textFour.setText(object.getTextFour());
-        footer.setText(object.getFooter());
+    public void updateClientDisplay(Client client) {
+        textClock.pause();
+        displayedClient = client;
+        for (int i = 0; i < clientObservableList.size(); i++) {
+            if (clientObservableList.get(i).isSameClient(client)) {
+                displayedClient = clientObservableList.get(i);
+                displayedClientIndex = i;
+            }
+        }
+        name.setText(client.getName().toString());
+        phone.setText(client.getPhone().toString());
+        email.setText(client.getEmail().toString());
+        country.setText(client.getCountry().getCountryName());
+        contractExpiryDate.setText("Expiry: " + client.getContractExpiryDate().displayValue);
+        noteTitle.setText("Notes");
+
+        Platform.runLater(() -> updateClientNotesDisplay(client.getClientNotesAsList()));
+        drawPaneBorder();
     }
 
     /**
-     * Updates the current content of the widget view box to the given content.
+     * Static factory.
```

Comment:
```suggestion
     * Static factory for use in GUI testing.
```
Actually, not sure if there's a need for this static factory since we can replace this with `new WidgetViewBox(FXCollections.emptyObservableList())` in our test functions